thrift: update (openssl3)

dkwo opened the pull request:
- I tested the changes in this PR: no
- I built this PR locally for my native architecture, (x86_64-musl)

required to build with openssl3 #37681

dkwo commented:
actually update it

Duncaen commented on srcpkgs/thrift/template:
```diff
@@ -1,24 +1,30 @@
+	-DWITH_JAVA=OFF \
+	-DWITH_JAVASCRIPT=OFF \
+	-DWITH_NODEJS=OFF \
+	-DWITH_PYTHON=OFF"
```
indent with single space.

Duncaen replied:
`\` is not needed in double quotes.

dkwo replied:
done.

kruceter commented on srcpkgs/thrift/template:
```diff
@@ -1,24 +1,30 @@
+version=0.17.0
+revision=1
+build_style=cmake
+configure_args="-DBUILD_SHARED_LIBS=ON
+ -DWITH_AS3=OFF
+ -DWITH_JAVA=OFF
+ -DWITH_JAVASCRIPT=OFF
+ -DWITH_NODEJS=OFF
+ -DWITH_PYTHON=OFF"
```
```suggestion
configure_args="-DBUILD_SHARED_LIBS=ON -DWITH_AS3=OFF -DWITH_JAVA=OFF
 -DWITH_JAVASCRIPT=OFF -DWITH_NODEJS=OFF -DWITH_PYTHON=OFF"
```

I do not think every option should be splattered one per line; correct me if I am wrong though.

dkwo commented:
> There is also a `sed` command in the template which should be replaced to `vsed`, but I myself am not sure who ever build on ppc64* these days.

I'll do as you say, but I did not check the ppc builds.

dkwo commented on srcpkgs/thrift/template:
```diff
@@ -1,24 +1,30 @@
+ -DWITH_NODEJS=OFF
+ -DWITH_PYTHON=OFF"
+hostmakedepends="flex"
+makedepends="boost-devel openssl-devel flex"
```
ok, let me try

dkwo commented on srcpkgs/thrift/template:
```diff
@@ -1,24 +1,30 @@
-distfiles="http://www-us.apache.org/dist/thrift/${version}/thrift-${version}.tar.gz"
-checksum=7ad348b88033af46ce49148097afe354d513c1fca7c607b59c33ebb6064b5179
+distfiles="https://github.com/apache/thrift/archive/v${version}.tar.gz"
+checksum=54c30b6ac27d62aee29e8d07831c1c785bf7ed8cd351e84130abe7d4f564335d
```
yes, github changed something two or more times the other day..

kruceter commented:
@Vaelatern, if you may?

paper42 commented on srcpkgs/thrift/template:
```diff
@@ -35,7 +41,6 @@ thrift-devel_package() {
 		vmove usr/include
-		vmove "usr/lib/*.a"
 		vmove "usr/lib/*.so"
 		vmove usr/lib/pkgconfig
```
```suggestion
		vmove usr/lib/cmake
		vmove usr/lib/pkgconfig
```

dkwo replied:
Done

dkwo commented:
This test was passing before on x86:
```
383/408 Test #383: TServerIntegrationTest ...........***Exception: SegFault  3.85 sec
terminate called after throwing an instance of 'std::system_error'
  what():  Resource temporarily unavailable
Running 13 test cases...
unknown location(0): fatal error: in "constructors/test_threaded_stress": signal: SIGABRT (application abort requested)
/builddir/thrift-0.17.0/lib/cpp/test/TServerIntegrationTest.cpp(373): last checkpoint: "test_threaded_stress" test entry
Thrift: Sun Feb  5 15:50:46 2023 TSocket::open() connect() <Host: 127.0.0.1 Port: 32931>: Connection refused
Thrift: Sun Feb  5 15:50:46 2023 TSocket::open() connect() <Host: 127.0.0.1 Port: 32931>: Connection refused
terminate called recursively
unknown location(0): fatal error: in "constructors/test_threadpool_factory": signal: SIGABRT (application abort requested)
/builddir/thrift-0.17.0/lib/cpp/test/TServerIntegrationTest.cpp(378): last checkpoint: "test_threadpool_factory" test entry
Thrift: Sun Feb  5 15:50:46 2023 TSocket::open() connect() <Host: 127.0.0.1 Port: 32931>: Connection refused
Thrift: Sun Feb  5 15:50:46 2023 TSocket::open() connect() <Host: 127.0.0.1 Port: 32931>: Connection refused
terminate called recursively
unknown location(0): fatal error: in "constructors/test_threadpool": signal: SIGABRT (application abort requested)
/builddir/thrift-0.17.0/lib/cpp/test/TServerIntegrationTest.cpp(393): last checkpoint: "test_threadpool" test entry
```
Could the failure be triggered by some change in the containers?

dkwo commented:
updated to 0.18, removed patches, now one glibc test fails:
```
17/44 Test #17: UnitTests ........................***Failed    1.32 sec
Running 60 test cases...
unknown location(0): fatal error: in "ToStringTest/locale_de_DE_floating_point_to_string": std::runtime_error: locale::facet::_S_create_c_locale name not valid
/builddir/thrift-0.18.1/lib/cpp/test/ToStringTest.cpp(57): last checkpoint: "locale_de_DE_floating_point_to_string" test entry
An error message from getaddrinfo on the console is expected:
Thrift: Fri Jun  2 15:23:52 2023 getaddrinfo() -> -2; Name or service not known

*** 1 failure is detected in the test module "thrift"
```

dkwo commented:
working now

dkwo commented:
@Duncaen Can this be merged?

Bnyro commented:
(also addresses https://github.com/void-linux/void-packages/issues/42403)
